byway/src/main/webapp/track-interests.js

```diff
@@ -59,6 +88,25 @@ function createButtonForPlace(place) {
   let button = document.createElement("button");
   button.innerText = place;
   button.addEventListener('click', () => updateStatus(place, button));
-  button.className = "btn";
+  button.className = "interestBtn";
   return button;
 }
+
+/**
+ * Gets all potential interests from server to load or sets the user's
+ * specific interests for their specific trip with tripQuery. userInterests
+ * may be any type that is convertible to an array via Array.from. It will
+ * be sent to the server as JSON in the post body.
+ * @param {String tripQuery} value for tripId
+ * @param {Array} [userInterests] interests selected by user
+ */
+function fetchPlaces(tripQuery, /* optional */ userInterests) {
+  const url = '/api/places?' + tripQuery;
```

Comment:
Prefer to use `URLSearchParams` to construct values for query parameters.

Reply:
Fixed